Improve documentation (#1588)
swiftlang/vscode-swift: userdocs/userdocs.docc/Articles/Reference/settings.md

vscode-swift provides various settings to configure its behaviour.

> > 💡 Tip: The Settings UI is common to all VS Code extensions. See the [VS Code documentation about personalizing VS Code](https://code.visualstudio.com/docs/getstarted/personalize-vscode) for a more in-depth overview.

Reviewer: Settings editor is preferred term I believe

Reviewer: maybe `See the [documentation about personalizing VS Code]` removing a redundant `VS Code`

## SourceKit-LSP

> [SourceKit-LSP](https://github.com/apple/sourcekit-lsp) is the language server used by the the Swift extension to provide symbol completion, jump to definition, etc. It is developed by Apple to provide Swift and C language support for any editor that supports the Language Server Protocol.

Reviewer 2: ```suggestion
[SourceKit-LSP](https://github.com/apple/sourcekit-lsp) is the language server used by the Swift extension to provide symbol completion, jump to definition, etc. It is developed by Apple to provide Swift and C language support for any editor that supports the Language Server Protocol.
```

### Background Indexing

Background Indexing was added as an experimental feature in Swift toolchain version 6.0 and then enabled by default in 6.1. This feature removes the need to do a build before getting code completion and diagnostics.

On startup, SourceKit-LSP will read your project information from your `Package.swift` and begin indexing your project automatically. All indexing results are cached in the `.build/index-build` folder within your workspace.

> To enable or disable this feature you can use the [`swift.sourcekit-lsp.backgroundIndexing`](vscode://settings/swift.sourcekit-lsp.backgroundIndexing) setting.

Reviewer 2: active voice
```suggestion
You can enable or disable this feature with the [`swift.sourcekit-lsp.backgroundIndexing`](vscode://settings/swift.sourcekit-lsp.backgroundIndexing) setting.
```
